app/src/main/java/com/github/se/eventradar/viewmodel/AuthenticationViewModel.kt

```diff
@@ -24,7 +24,7 @@ import kotlinx.coroutines.launch
 import kotlinx.coroutines.runBlocking
 
 @HiltViewModel
-class LoginViewModel @Inject constructor(private val userRepository: IUserRepository) :
+open class LoginViewModel @Inject constructor(private val userRepository: IUserRepository) :
```

Comment:
I understand that making a class open allows it to be subclassed, but could you please clarify the specific reason for making LoginViewModel an open class in this context?

Reply:
Nice catch! I had been playing around to optimise the duplication of code between the LoginViewModel and the ProfileViewModel. It should be a private class you're right